src/redux/rockets/rocketsSlice.js

```diff
@@ -0,0 +1,27 @@
+// rocketsSlice.js
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export const fetchRockets = createAsyncThunk('rockets/fetchRockets', async () => {
+  const response = await axios.get('https://api.spacexdata.com/v4/rockets');
+  return response.data.map((rocket) => ({ ...rocket, reserved: false }));
+});
+
+const rocketsSlice = createSlice({
+  name: 'rockets',
+  initialState: [],
+  reducers: {
+    toggleReservation: (state, action) => {
+      const rocket = state.find((rocket) => rocket.id === action.payload);
+      if (rocket) {
+        rocket.reserved = !rocket.reserved;
+      }
+    },
+  },
```

Comment:
- [ OPTOINAL] I suggest using `filter()` or `map()` to efficiently update the state. Like this:point_down: 
```javascript
 const newState = state.map(rocket => {
    if(rocket.id !== id) 
        return rocket;
    return { ...rocket, reserved: false };
});
```